inc/Engine/License/Renewal.php

```diff
@@ -0,0 +1,304 @@
+<?php
+
+namespace WP_Rocket\Engine\License;
+
+use WP_Rocket\Abstract_Render;
+use WP_Rocket\Engine\License\API\Pricing;
+use WP_Rocket\Engine\License\API\User;
+
+class Renewal extends Abstract_Render {
+	/**
+	 * Pricing instance
+	 *
+	 * @var Pricing
+	 */
+	private $pricing;
+
+	/**
+	 * User instance
+	 *
+	 * @var User
+	 */
+	private $user;
+
+	/**
+	 * Instantiate the class
+	 *
+	 * @param Pricing $pricing       Pricing instance.
+	 * @param User    $user          User instance.
+	 * @param string  $template_path Path to the views.
+	 */
+	public function __construct( Pricing $pricing, User $user, $template_path ) {
+		parent::__construct( $template_path );
+
+		$this->pricing = $pricing;
+		$this->user    = $user;
+	}
+
+	/**
+	 * Displays the renewal banner for users expiring in less than 30 days
+	 *
+	 * @since 3.7.5
+	 *
+	 * @return void
+	 */
+	public function display_renewal_soon_banner() {
+		if ( rocket_get_constant( 'WP_ROCKET_WHITE_LABEL_ACCOUNT' ) ) {
+			return;
+		}
+
+		if ( $this->user->is_license_expired() ) {
+			return;
+		}
+
+		if ( ! $this->is_expired_soon() ) {
+			return;
+		}
+
+		$data = [
+			'discount_percent' => $this->get_discount_percent(),
+			'discount_price'   => $this->get_discount_price(),
+			'countdown'        => $this->get_countdown_data(),
+			'renewal_url'      => $this->user->get_renewal_url(),
+		];
+
+		echo $this->generate( 'renewal-soon-banner', $data ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	}
+
+	/**
+	 * Displays the renewal banner for expired users
+	 *
+	 * @since 3.7.5
+	 *
+	 * @return void
+	 */
+	public function display_renewal_expired_banner() {
+		if ( rocket_get_constant( 'WP_ROCKET_WHITE_LABEL_ACCOUNT' ) ) {
+			return;
+		}
+
+		if ( ! $this->user->is_license_expired() ) {
+			return;
+		}
+
+		if ( false !== get_transient( 'rocket_renewal_banner_' . get_current_user_id() ) ) {
+			return;
+		}
+
+		$data = [
+			'discount_percent' => $this->get_discount_percent(),
+			'discount_price'   => $this->get_discount_price(),
+			'renewal_url'      => $this->user->get_renewal_url(),
+		];
+
+		echo $this->generate( 'renewal-expired-banner', $data ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	}
+
+	/**
+	 * AJAX callback to dismiss the renewal banner for expired users
+	 *
+	 * @since 3.7.5
+	 *
+	 * @return void
+	 */
+	public function dismiss_renewal_expired_banner() {
+		check_ajax_referer( 'rocket-ajax', 'nonce', true );
+
+		if ( ! current_user_can( 'rocket_manage_options' ) ) {
+			return;
+		}
+
+		$user = get_current_user_id();
+
+		if ( false !== get_transient( "rocket_renewal_banner_{$user}" ) ) {
+			return;
+		}
+
+		set_transient( "rocket_renewal_banner_{$user}", 1, MONTH_IN_SECONDS );
```

Comment:
Tiny performance improvement:

```suggestion
		$transient = 'rocket_renewal_banner_' . get_current_user_id();

		if ( false !== get_transient( $transient ) ) {
			return;
		}

		set_transient( $transient, 1, MONTH_IN_SECONDS );
```

Why? Avoids doing the string concatenation twice.